src/modules/library.ts

```diff
@@ -26,303 +33,605 @@ export enum Buttons {
+    data: ElementDataObject;
+    maxAttempts?: number;
     delayInMillis?: number;
-    postRetries?: number;
+    httpRetries?: number;
   }) {
-    let functionRetries = 0;
-    while (functionRetries < maxRetries) {
-      const uiLayoutResponse = await this.driver.getUIElementError({ data, retries: postRetries });
-      if (uiLayoutResponse.status !== 200) functionRetries++;
-      else return true;
-      await sleep(delayInMillis);
+    let attempts = 0;
+    while (attempts < maxAttempts) {
+      try {
+        const response = await this.driver.getUIElement({ data, retries: httpRetries });
+        if (response) {
+          return true;
+        } else {
+          attempts++;
+          await sleep(delayInMillis);
+        }
+      } catch (error) {
+        console.log(error);
+        attempts++;
+        await sleep(delayInMillis);
+      }
     }
     return false;
   }
 
   /** Alias for verifyIsScreenLoaded with a more intuitive name */
-  async verifyIsElementOnScreen({
+  verifyIsElementOnScreen({
     data,
-    maxRetries = 10,
-    delayInMillis = 1000,
-    postRetries = 3,
+    maxAttempts = 10,
+    delayInMillis = this.retryDelayInMillis,
+    httpRetries = this.retries,
   }: {
-    data: elementDataObject;
-    maxRetries?: number;
+    data: ElementDataObject;
+    maxAttempts?: number;
     delayInMillis?: number;
-    postRetries?: number;
+    httpRetries?: number;
   }) {
-    return this.verifyIsScreenLoaded({ data, maxRetries, delayInMillis, postRetries });
+    return this.verifyIsScreenLoaded({ data, maxAttempts, delayInMillis, httpRetries });
   }
 
   /** Simulates the press and release of the specified key. */
   async pressBtn({
     keyPress,
-    delayInMillis = 2000,
-    retries = 3,
+    delayInMillis = this.pressDelayInMillis,
+    retries = this.retries,
+    params,
   }: {
     keyPress: string;
     delayInMillis?: number;
-    retries?: 3;
+    retries?: number;
+    params?: Params;
   }) {
+    const response = this.driver.sendKeypress({ keyPress, retries, params });
     await sleep(delayInMillis);
-    return this.driver.sendKeypress({ keyPress, retries });
+    return response;
   }
 
   /** Simulates the press and release of each letter in a word. */
   async sendWord({
     word,
-    delayInMillis = 2000,
-    retries = 3,
+    delayInMillis = this.pressDelayInMillis,
+    retries = this.retries,
+    params,
   }: {
     word: string;
     delayInMillis?: number;
     retries?: number;
+    params?: Params;
   }) {
-    await sleep(delayInMillis);
-    const wordResponse: { [key: string]: nullValueResponse }[] = [];
+    const sequence: string[] = [];
     for (let charIndex = 0; charIndex < word.length; charIndex++) {
-      await sleep(500);
       const key = word.charAt(charIndex);
-      const value = await this.driver.sendKeypress({ keyPress: 'LIT_' + key, retries });
-      wordResponse[charIndex] = { [key]: value };
+      sequence.push(`LIT_${key}`);
```

Comment:
you'll want to do `LIT_${encodeURIComponent(key)}`

Reply:
I swore that this was working when I had tried it, but I'll double-check this.

Reply:
Axios automatically does this, you shouldn't need to do it manually

Reply:
Sam is right, it was working as expected with the `encodeURIComponent`...
But your review made me realize I did something else wrong (the `.forEach` with sequences was doing weirdness but switching for a regular `for` fixed it)
c871193